src/storage/garbage_collector.cpp

```diff
@@ -87,32 +122,111 @@ uint32_t GarbageCollector::ProcessUnlinkQueue() {
       // to safely free the txn
       txns_to_deallocate_.push_front(txn);
       txns_processed++;
-    } else if (transaction::TransactionUtil::NewerThan(oldest_txn, txn->TxnId().load())) {
-      // Safe to garbage collect.
+    } else {
+      // This is a txn that may or may not be visible to any running txns. Proceed with unlinking its UndoRecords
+      // with an Interval GC approach
+      bool all_unlinked = true;
```

Comment:
`all_unlinked` looks suspiciously like a legacy from our previous implementations. Is there a real failure scenario where you need this?

Reply:
We still need this logic to know when all of the transaction's undo records have been unlinked. #325 eliminates this by only processing a transaction for unlinking when it is older than all the active txns. Here we do not and can not make this assumption (Interval GC).